### PR TITLE
Cherry-pick commits related to migration away from ctx param in partials

### PR DESCRIPTION
**Pull request by segiddins**

### CODE REVIEW

File: test/starlark_tests/verifier_scripts/archive_contents_test.sh [hunk 1]
```diff
@@ -139,7 +139,10 @@ if [[ -n "${BINARY_TEST_FILE-}" ]]; then
 
   if [[ -n "${CODESIGN_INFO_CONTAINS-}" || -n "${CODESIGN_INFO_NOT_CONTAINS-}" ]]; then
     codesign_output="$(mktemp "${TMPDIR:-/tmp}/codesign_output.XXXXXX")"
-    codesign --display --verbose=4 "$path" &> "$codesign_output"
+    if ! codesign --display --verbose=4 "$path" &> "$codesign_output"; then
+      fail "Expected codesign --display to pass$newline$(cat $codesign_output)"
+    fi
+    cat "$codesign_output"
```
Comment (thii) [inline test/starlark_tests/verifier_scripts/archive_contents_test.sh:145]: Is this intentional? It'd be printed out twice if the test fails.

Comment (segiddins) [reply test/starlark_tests/verifier_scripts/archive_contents_test.sh:145]: unintentional

**Review by keith — APPROVED** [review]

**Comment by keith** [comment]
@thii wanna look over this again since most of the conflicts were related to changes you made?

**Review by thii — APPROVED** [review]
LGTM. Thanks for taking care of them.

**Pull request merged**